File: Leetcode-Pattern-problems-main/Leetcode-Pattern-problems-main/Trees/height_of_tree.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) :
        self.val = val
        self.left = left
        self.right = right


def height(node):
    if node is None:
        return 0
    return 1 + max(height(node.left),height(node.right))


root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)
root.left.left = TreeNode(4)
root.left.right = TreeNode(5)
root.right.left = TreeNode(6)
root.right.right = TreeNode(7)
root.right.right.right = TreeNode(8)

print(height(root))

File: Binary tree/height_of_tree.cpp
#include<iostream>
#include<vector>
using namespace std;
class node{
    public :
    int data ;
    node* left;
    node* right;
    node(int val){
        data = val;
        left = right = NULL;
    }
};
static int idx = -1;
node* buildtree(vector<int> preorder){
    idx++;
    if(preorder[idx]==-1){
        return NULL;
    }
    node* root = new node(preorder[idx]);
    root->left = buildtree(preorder);
    root->right= buildtree(preorder);
    return root;
};
int height(node* root){
    if(root == NULL) return  0 ;
    int left;
    left = height(root->left);
    int right;
    right = height(root->right); 
    return max(left,right) + 1;
}
int main(){
    vector<int> preorder={1,2,-1,-1,3,4,-1,-1,5,-1,-1};
    node* root = buildtree(preorder);
    cout<<height(root)<<" ";
}